claude_scripts/enhanced_learn.py

```
@@ -0,0 +1,129 @@
+    """Enhanced learn function with pattern detection and storage"""
+
+    # Ensure memory directory exists
+    MEMORY_DIR.mkdir(parents=True, exist_ok=True)
+
+    # Load existing memory
+    memory = {}
+    if MEMORY_FILE.exists():
+        try:
+            with open(MEMORY_FILE) as f:
+                memory = json.load(f)
+        except:
```

Ruff (0.14.4)

Do not use bare `except`

(E722)